sdk/tests/wallet/address_generation.rs

```diff
@@ -0,0 +1,199 @@
+    }
+    let wallet = wallet_builder.finish().await?;
+
+    let address = wallet.generate_ed25519_address(0, 0, None).await?;
+
+    assert_eq!(
+        address.to_bech32_unchecked("smr"),
+        // Address generated with bip32 path: [44, 4218, 0, 0, 0].
+        "smr1qrpwecegav7eh0z363ca69laxej64rrt4e3u0rtycyuh0mam3vq3ulygj9p"
+    );
+
+    tear_down(storage_path)
+}
+
+#[cfg(feature = "stronghold")]
+#[tokio::test]
+async fn wallet_address_generation_stronghold() -> Result<()> {
+    let storage_path = "test-storage/wallet_address_generation_stronghold";
+    setup(storage_path)?;
+
+    iota_stronghold::engine::snapshot::try_set_encrypt_work_factor(0).unwrap();
+
+    let secret_manager = StrongholdSecretManager::builder()
+        .password("some_hopefully_secure_password".to_owned())
+        .build("test-storage/wallet_address_generation_stronghold/test.stronghold")?;
+    secret_manager
+        .store_mnemonic(Mnemonic::from(DEFAULT_MNEMONIC.to_string()))
+        .await?;
+
+    let client_options = ClientOptions::new().with_node(NODE_LOCAL)?;
+    #[allow(unused_mut)]
+    let mut wallet_builder = Wallet::builder()
+        .with_secret_manager(SecretManager::Stronghold(secret_manager))
+        .with_client_options(client_options)
+        .with_coin_type(IOTA_COIN_TYPE);
+    #[cfg(feature = "storage")]
+    {
+        wallet_builder = wallet_builder.with_storage_path(storage_path);
+    }
+    let wallet = wallet_builder.finish().await?;
+
+    let address = wallet.generate_ed25519_address(0, 0, None).await?;
+
+    assert_eq!(
+        address.to_bech32_unchecked("smr"),
+        // Address generated with bip32 path: [44, 4218, 0, 0, 0].
+        "smr1qrpwecegav7eh0z363ca69laxej64rrt4e3u0rtycyuh0mam3vq3ulygj9p"
+    );
+
+    tear_down(storage_path)
+}
+
+#[tokio::test]
+#[cfg(feature = "ledger_nano")]
+#[ignore = "requires ledger nano instance"]
+async fn wallet_address_generation_ledger() -> Result<()> {
+    let storage_path = "test-storage/wallet_address_generation_ledger";
+    setup(storage_path)?;
+
+    let client_options = ClientOptions::new().with_node(NODE_LOCAL)?;
+    let mut secret_manager = LedgerSecretManager::new(true);
+    secret_manager.non_interactive = true;
+
+    #[allow(unused_mut)]
+    let mut wallet_builder = Wallet::builder()
+        .with_secret_manager(SecretManager::LedgerNano(secret_manager))
+        .with_client_options(client_options)
+        .with_coin_type(IOTA_COIN_TYPE);
+
+    #[cfg(feature = "storage")]
+    {
+        wallet_builder = wallet_builder.with_storage_path(storage_path);
+    }
+    let wallet = wallet_builder.finish().await?;
+
+    let address = wallet.generate_ed25519_address(0, 0, None).await?;
+
+    assert_eq!(
+        address.to_bech32_unchecked("smr"),
+        // Address generated with bip32 path: [44, 4218, 0, 0, 0].
+        // This address was generated with a MnemonicSecretManager and the ledger simulator mnemonic.
```

Comment:
Do you have the mnemonic for it? Would be more helpful than this comment

Reply:
Sure